### ISSUE TITLE
Some way of code returning what context it is in?

### ISSUE THREAD

**Issue by BruceEdmonds** [NetLogo/NetLogo#882]
Due to the woeful lack of debugging facilities in NetLogo (grumble grumble) it is useful to be able to add commands into the code to optionally report all sorts of things about the current state.

Unfortunately such procedures can not be universal due to not being able to use "self" in the observer.

Even if one writes something like:
     carefully [set id (word self)] [set id "Observer"]

The Netlogo syntax checker does not allow this.

Can we have either:
1. a primitive that returns the name of the entity executing the code, including the observer
OR
2. allow self in the Observer context but  return "nobody" or similar 

Thanks.

**Comment by SethTisue** [comment]
I agree this is a hole in the language. I'm not sure what the best way to address it would be.

Note also that `show` works for all agent types and prepends the output with a string such as `observer:` or `turtle 3`. This addresses the single most common use case.

For other use cases, here's a version with `carefully` that does work:

```netlogo
to-report agent-string
  let t task [(word self)]
  let result "observer"
  carefully [ set result runresult t ] [ ]
  report result
end
```

**Comment by SethTisue** [comment]
P.S. If you can be of any assistance in helping us obtain funding to build better debugging facilities for NetLogo, please let Uri Wilensky know.

**Comment by qiemem** [comment]
Note this was also discussed in this SO question: http://stackoverflow.com/a/33814510/145080

**Comment by Idloj** [comment]
Also found here: https://groups.google.com/forum/#!topic/netlogo-devel/Nw-S_QyIafE

**Comment by SethTisue** [comment]
just `runresult task [(word self)]` by itself works in NetLogo 5.2.x on the JVM, but only because of a bug, and it does not work in NetLogo Web; see #888 for details. so I would recommend using `carefully`
